runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -230,6 +434,89 @@ CachedAOTMethod::create(const AOTCacheClassChainRecord *definingClassChainRecord
                                     records, code, codeSize, data, dataSize);
    }
 
+CachedAOTMethod *
+CachedAOTMethod::read(FILE *f,
+                      const Vector<AOTCacheClassLoaderRecord *> &classLoaderRecords,
+                      const Vector<AOTCacheClassRecord *> &classRecords,
+                      const Vector<AOTCacheMethodRecord *> &methodRecords,
+                      const Vector<AOTCacheClassChainRecord *> &classChainRecords,
+                      const Vector<AOTCacheWellKnownClassesRecord *> &wellKnownClassesRecords,
+                      const Vector<AOTCacheAOTHeaderRecord *> &aotHeaderRecords)
+   {
+   SerializedAOTMethod header;
+   if (1 != fread(&header, sizeof(header), 1, f))
+      return NULL;
+
+   if ((!header.isValid()) ||
+       (header.definingClassChainId() >= classChainRecords.size()) ||
+       (header.aotHeaderId() >= aotHeaderRecords.size()))
+      return NULL;
+
+   auto record = new (AOTCacheRecord::allocate(size(header.numRecords(), header.codeSize(), header.dataSize())))
+                     CachedAOTMethod(classChainRecords[header.definingClassChainId()]);
+   memcpy(&record->_data, &header, sizeof(header));
+
+   if (1 != fread(record->data().offsets(), sizeof(SerializedSCCOffset) * header.numRecords() + header.codeSize() + header.dataSize(), 1, f))
+      {
+      AOTCacheRecord::free(record);
```

Comment:
I would actually suggest using `goto error;` to factor this out, unless the JIT coding style prohibits that.

Reply:
I have seen `goto` in the code, so I don't think we prohibit it.